cmd/tuf/app/add-delegation.go

```diff
@@ -44,28 +44,37 @@ func (f *keysFlag) Set(value string) error {
 	return nil
 }
 
+type DelegationOptions struct {
+	Directory   string
+	Name        string
+	Path        string
+	Terminating bool
+	KeyRefs     []string
+	Threshold   int
+	Targets     string
+}
+
 func AddDelegation() *ffcli.Command {
 	var (
-		flagset     = flag.NewFlagSet("tuf add-delegation", flag.ExitOnError)
-		repository  = flagset.String("repository", "", "path to the staged repository")
-		name        = flagset.String("name", "", "name of the delegatee")
-		keys        = keysFlag{}
-		path        = flagset.String("path", "", "path for the delegation")
-		terminating = flagset.Bool("terminating", false, "indicated whether the delegation is terminating")
```

Comment:
+1 thank you for removing terminating option, and removing the option for path setting via CLI